packages/protocol/contracts/governance/VestingInstance.sol

```diff
@@ -0,0 +1,356 @@
+
+        require(releasableAmount > 0, "No unreleased tokens are due for withdraw");
+
+        mCurrentlyReleased = mCurrentlyReleased.add(releasableAmount);
+
+        getGoldToken().safeTransfer(mBeneficiary, releasableAmount);
+
+        emit VestingWithdrawn(msg.sender, releasableAmount, block.timestamp);
+    }
+
+    /**
+     * @notice Allows only the revoker to revoke the vesting. Gold already vested
+     * remains in the contract, the rest is returned to the _refundDestination.
+     * @param revokeTime the revocation timestamp
+     * @dev revokeTime the revocation timestamp. If is less than the current block timestamp, it is set equal
+     */
+    function revoke(revokeTime) external onlyRevoker {
+        require(mRevocable, "Revoking is not allowed");
+        require(!mRevoked, "Vesting already revoked");
+
+        uint256 revokeTimestamp = revokeTime > block.timestamp ? revokeTime : block.timestamp;
+
+        uint256 balance = getGoldToken().balanceOf(address(this));
+        uint256 releasableAmount = _getReleasableAmount(revokeTimestamp);
+        uint256 refund = balance.sub(releasableAmount);
+
+        mRevoked = true;
+        mRevokeTime = revokeTimestamp;
+
+        getGoldToken().transfer(mRefundDestination, refund);
+
+        emit VestingRevoked(msg.sender, mRefundDestination, refund, mRevokeTime);
+    }
+
+    /**
+     * @dev Calculates the amount that has already vested but hasn't been withdrawn (released) yet.
+     * @param timestamp the timestamp at which the calculate the releasable amount
+     */
+    function _getReleasableAmount(uint256 timestamp) public view returns (uint256) {
+        return _calculateFreeAmount(timestamp).sub(mCurrentlyReleased);
+    }
+
+    /**
+     * @dev Calculates the amount that has already vested.
+     * @param timestamp the timestamp at which the calculate the already vested amount
+     */
+    function _calculateFreeAmount(uint256 timestamp) private view returns (uint256) {
+        uint256 currentBalance = getGoldToken().balanceOf(address(this));
+        uint256 totalBalance = currentBalance.add(mCurrentlyReleased);
+
+        if (timestamp < mCliffStartTime) {
+            return 0;
+        }
+        if (timestamp >= mVestingStartTime.add( mVestingPeriods.mul(mVestingPeriodSec) ) || mRevoked) {
+            return totalBalance;
+        }
+        
+        uint256 gradient = (timestamp.sub(mVestingStartTime)).div(mVestingPeriodSec);
+        return  ( (currentBalance.mul(gradient)).mul(mVestAmountPerPeriod) ).div(totalBalance);
+    }
+
+    /**
+     * @notice A wrapper func for the lock gold method
+     * @param value the value to gold to be locked
+     * @return True if the transaction succeeds.
+     * @dev To be called only by the beneficiary of the vesting
+     */
+    function lockGold(uint256 value) external onlyBeneficiary returns (bool) {
+
+      // the beneficiary may not lock more than the vesting has currently released
```

Comment:
Actually we're totally fine with the beneficiary locking all of the gold, so this function should basically just be the call to `lock`

Reply:
got it, changed